srcpkgs/openjdk9-bootstrap/patches/fix-pointer-comparison-gcc12.patch

```diff
@@ -0,0 +1,22 @@
+--- a/hotspot/src/share/vm/opto/lcm.cpp
++++ b/hotspot/src/share/vm/opto/lcm.cpp
+@@ -60,7 +60,7 @@
+ // Check whether val is not-null-decoded compressed oop,
+ // i.e. will grab into the base of the heap if it represents NULL.
+ static bool accesses_heap_base_zone(Node *val) {
+-  if (Universe::narrow_oop_base() > 0) { // Implies UseCompressedOops.
++  if (Universe::narrow_oop_base()) { // Implies UseCompressedOops.
```

Comment:
If the intention is compare to `nullptr` please explicit, or write the patch description.
As of it's, I assume `Universe::narrow_oop_base()` returns an integer.

Reply:
`Universe::narrow_oop_base()` returns a char pointer, that's why the comparison is invalid (since it is comparing a pointer and an integer).